src/test/java/com/uber/cadence/worker/WorkerStressTests.java

```diff
@@ -0,0 +1,201 @@
+  private class TestEnvironmentWrapper {
+
+    private TestWorkflowEnvironment testEnv;
+    private Worker.Factory factory;
+
+    public TestEnvironmentWrapper(Worker.FactoryOptions options) {
+      if (options == null) {
+        options = new Worker.FactoryOptions.Builder().Build();
+      }
+      factory = new Worker.Factory(DOMAIN, options);
+      TestEnvironmentOptions testOptions =
+          new TestEnvironmentOptions.Builder().setDomain(DOMAIN).setFactoryOptions(options).build();
+      testEnv = TestWorkflowEnvironment.newInstance(testOptions);
+    }
+
+    private Worker.Factory getWorkerFactory() {
+      return useExternalService ? factory : testEnv.getWorkerFactory();
+    }
+
+    private WorkflowClient getWorkflowClient() {
+      return useExternalService ? WorkflowClient.newInstance(DOMAIN) : testEnv.newWorkflowClient();
+    }
+
+    private void close() {
+      factory.shutdown(Duration.ofSeconds(1));
+      testEnv.close();
+    }
+  }
+
+  public static class WorkflowParams {
+
+    public int ChainSequence;
+    public int ConcurrentCount;
+    public String TaskListName;
+    public int PayloadSizeBytes;
+    public Duration CadenceSleep;
+  }
+
+  public interface ActivitiesWorkflow {
+
+    @WorkflowMethod()
+    String execute(WorkflowParams params);
+  }
+
+  public static class ActivitiesWorkflowImpl implements ActivitiesWorkflow {
+    private static final Logger log = LoggerFactory.getLogger(ActivitiesWorkflowImpl.class);
+
+    @Override
+    public String execute(WorkflowParams params) {
+      SleepActivity activity =
+          Workflow.newActivityStub(
+              SleepActivity.class,
+              new ActivityOptions.Builder()
+                  .setTaskList(params.TaskListName)
+                  .setScheduleToStartTimeout(Duration.ofMinutes(1))
+                  .setStartToCloseTimeout(Duration.ofMinutes(1))
+                  .setHeartbeatTimeout(Duration.ofSeconds(20))
+                  .build());
+
+      for (int i = 0; i < params.ChainSequence; i++) {
+        List<Promise<Void>> promises = new ArrayList<>();
+        for (int j = 0; j < params.ConcurrentCount; j++) {
+          byte[] bytes = new byte[params.PayloadSizeBytes];
+          new Random().nextBytes(bytes);
+          Promise<Void> promise = Async.procedure(activity::sleep, i, j, bytes);
+          promises.add(promise);
+        }
+
+        for (Promise<Void> promise : promises) {
+          promise.get();
+        }
+
+        Workflow.sleep(params.CadenceSleep);
+      }
+      return "I'm done";
+    }
+  }
+
+  public interface SleepActivity {
+
+    @ActivityMethod()
+    void sleep(int chain, int concurrency, byte[] bytes);
+  }
+
+  public static class ActivitiesImpl implements SleepActivity {
+    private static final Logger log = LoggerFactory.getLogger("sleep-activity");
+
+    @Override
+    public void sleep(int chain, int concurrency, byte[] bytes) {
+      log.info("sleep called");
```

Comment:
I'm confused about purpose and name of this activity as it never sleeps.

Reply:
renamed it to someActivity. It's purpose is to simply to be triggered, consume threads and generate events